node/src/components/small_network.rs

```diff
@@ -1028,22 +960,18 @@ where
                 effects
             }
             Event::LinearChainAnnouncement(LinearChainAnnouncement::BlockAdded(block)) => {
-                // On switch blocks, we need to update our validator sets.
-                if block.header().is_switch_block() {
-                    let era_id = block.header().era_id();
-
-                    if era_id > self.highest_era_seen {
-                        self.highest_era_seen = era_id;
-                        self.handle_active_era_change(effect_builder, era_id)
-                    } else {
-                        debug!(highest_era_seen=%self.highest_era_seen, %era_id,
-                               "ignoring era, as it is not the highest seen");
-                        Effects::new()
-                    }
-                } else {
-                    trace!("ignoring non-switch block");
-                    Effects::new()
-                }
+                let _era_id = block.header().era_id();
+                if let Some(next_era_validators) =
+                    block.take_header().maybe_take_next_era_validator_weights()
+                {
+                    let upcoming_validators: HashSet<PublicKey> =
+                        next_era_validators.into_keys().collect();
+                    self.outgoing_limiter
+                        .update_validators(Default::default(), upcoming_validators.clone());
+                    self.incoming_limiter
+                        .update_validators(Default::default(), upcoming_validators);
```

Comment:
This is now changing the logic. Previously, on every switch block it would upcoming validators for 2 eras in the future (the immediate era after the switch block and validators after that era, since Auction contract has already determined who those validators will be). New logic only updates the next era validators so we don't get the benefit of prioritizing soon-to-be-validators nodes' traffic. /cc @marc-casperlabs

Reply:
@goral09 is correct. However that was the *intended* behavior. Sadly, in the past, the function that is supposed to determine the `n+2` era validators would always return an empty set due to a bug/shortcoming in the function itself, but fixing that bug was outside the scope of implementing the upstream limiter itself.

So the code in small networking was written in the hopes of that bug being fixed and "future proof" in a way, but as far as I remember the fix has not materialized yet.

We are about to change this section of code once again, so I am fine with the changes here for now. We will replace it before 1.4, and additionally it should not degrade the _actual_ behavior at the moment.

Pretty good catch though nonetheless :+1:

Reply:
@marc-casperlabs #1469 was supposed to be the fix you mention :wink:

Reply:
This will be handled in an upcoming PR